### PR TITLE
Redirect stdout/stderr to log file

### PR DESCRIPTION
**Pull request by yixinglu**
for sanitizer validation in graph CI

### CODE REVIEW

File: conf/nebula-metad.conf.default [hunk 1]
```diff
@@ -13,6 +13,11 @@
 --v=0
 # Maximum seconds to buffer the log messages
 --logbufsecs=0
+# Whether to redirect stdout and stderr to separate output files
+--redirect_stdout=true
+# Destination filename of stdout and stderr, which will also reside in log_dir.
+--stdout_log_file=metad-stdout.log
```
Comment (Shylock-Hg) [inline conf/nebula-metad.conf.default:19]: logs/metad-stdout.log?

Comment (yixinglu) [reply conf/nebula-metad.conf.default:19]: `setupLogging` will join the `FLAGS_log_dir` and `FLAGS_stdout_log_file` paths

**Review by critical27 — APPROVED** [review]

File: src/daemons/SetupLogging.cpp [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+    auto dup = [](const std::string &filename, FILE *stream) -> Status {
+        auto path = FLAGS_log_dir + "/" + filename;
+        auto fd = ::open(path.c_str(), O_WRONLY | O_APPEND | O_CREAT, 0644);
+        if (fd == -1) {
+            return Status::Error(
+                "Failed to create or open `%s': %s", path.c_str(), ::strerror(errno));
+        }
+        if (::dup2(fd, ::fileno(stream)) == -1) {
+            return Status::Error(
```
Comment (critical27) [inline src/daemons/SetupLogging.cpp:31]: close fd?

**Review by critical27 — APPROVED** [review]
Generally LGTM

File: src/daemons/SetupLogging.cpp [hunk 3]
```diff
@@ -0,0 +1,41 @@
+/* Copyright (c) 2018 vesoft inc. All rights reserved.
```
Comment (laura-ding) [inline src/daemons/SetupLogging.cpp:1]: 2020

**Review by CPWstatic — APPROVED** [review]

File: src/daemons/SetupLogging.cpp [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+DECLARE_string(log_dir);
+
+DEFINE_bool(redirect_stdout, true, "Whether to redirect stdout and stderr to separate files");
+DEFINE_string(stdout_log_file, "stdout.log", "Destination filename of stdout");
+DEFINE_string(stderr_log_file, "stderr.log", "Destination filename of stderr");
+
+using nebula::Status;
+
+Status setupLogging() {
```
Comment (laura-ding) [inline src/daemons/SetupLogging.cpp:18]: the same as graphd, why not move it to common?

**Pull request merged**